vcd/resource_vcd_vcenter.go

```diff
@@ -0,0 +1,274 @@
+			"version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("Version of %s", labelVirtualCenter),
+			},
+			"uuid": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("%s UUID", labelVirtualCenter),
+			},
+		},
+	}
+}
+
+func getTmVcenterType(d *schema.ResourceData) (*types.VSphereVirtualCenter, error) {
+	t := &types.VSphereVirtualCenter{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Url:         d.Get("url").(string),
+		Username:    d.Get("username").(string),
+		Password:    d.Get("password").(string),
+		IsEnabled:   d.Get("is_enabled").(bool),
+	}
+
+	return t, nil
+}
+
+func setTmVcenterData(d *schema.ResourceData, v *govcd.VCenter) error {
+	if v == nil || v.VSphereVCenter == nil {
+		return fmt.Errorf("nil object for %s", labelVirtualCenter)
+	}
+
+	dSet(d, "name", v.VSphereVCenter.Name)
+	dSet(d, "description", v.VSphereVCenter.Description)
+	dSet(d, "url", v.VSphereVCenter.Url)
+	dSet(d, "username", v.VSphereVCenter.Username)
+	// dSet(d, "password", v.VSphereVCenter.Password) // password is never returned,
+	dSet(d, "is_enabled", v.VSphereVCenter.IsEnabled)
+
+	dSet(d, "has_proxy", v.VSphereVCenter.HasProxy)
+	dSet(d, "is_connected", v.VSphereVCenter.IsConnected)
+	dSet(d, "mode", v.VSphereVCenter.Mode)
+	dSet(d, "listener_state", v.VSphereVCenter.ListenerState)
+	dSet(d, "cluster_health_status", v.VSphereVCenter.ClusterHealthStatus)
+	dSet(d, "version", v.VSphereVCenter.VcVersion)
+	dSet(d, "uuid", v.VSphereVCenter.Uuid)
+
+	d.SetId(v.VSphereVCenter.VcId)
+
+	return nil
+}
+
+func resourceVcdVcenterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	c := crudConfig[*govcd.VCenter, types.VSphereVirtualCenter]{
+		entityLabel:      labelVirtualCenter,
+		getTypeFunc:      getTmVcenterType,
+		stateStoreFunc:   setTmVcenterData,
+		createFunc:       vcdClient.CreateVcenter,
+		resourceReadFunc: resourceVcdVcenterRead,
+		// certificate should be trusted for the vCenter to work
+		preCreateHooks: []schemaHook{autoTrustHostCertificate("url", "auto_trust_certificate")},
+	}
+	return createResource(ctx, d, meta, c)
+}
+
+func resourceVcdVcenterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if !d.HasChangeExcept("refresh_on_read") {
+		return nil
+	}
+
+	vcdClient := meta.(*VCDClient)
+	c := crudConfig[*govcd.VCenter, types.VSphereVirtualCenter]{
+		entityLabel:      labelVirtualCenter,
+		getTypeFunc:      getTmVcenterType,
+		getEntityFunc:    vcdClient.GetVCenterById,
+		resourceReadFunc: resourceVcdVcenterRead,
+	}
+
+	return updateResource(ctx, d, meta, c)
+}
+
+func resourceVcdVcenterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	// TODO: TM: remove this block and use the commented one within crudConfig below.
```

Comment:
Would like to understand better, what is the a reason to not do it right now instead of later.

Reply:
This causes inconsistent Terraform plans. The "GetById" function returns host with port at the end `https://HOST:**443**` while the GetByName returns `https://HOST`